Index: include\collision.h
===================================================================
--- include\collision.h
+++ include\collision.h
@@ -20,8 +20,10 @@
 
 extern UBYTE tile_hit_x;
 extern UBYTE tile_hit_y;
 
+extern UBYTE scene_transition_enabled;
+
 /**
  * Check if point is within positioned bounding box.
  *
  * @param bb Pointer to bounding box
@@ -29,11 +31,11 @@
  * @param point Pointer to position to look for within bounding box
  * @return Point is within bounding box
  */
 inline UBYTE bb_contains(rect16_t *bb, upoint16_t *offset, upoint16_t *point) {
-    if ((point->x < offset->x + bb->left) || 
+    if ((point->x < offset->x + bb->left) ||
         (point->x > offset->x + bb->right)) return FALSE;
-    if ((point->y < offset->y + bb->top) || 
+    if ((point->y < offset->y + bb->top) ||
         (point->y > offset->y + bb->bottom)) return FALSE;
     return TRUE;
 }
 
@@ -47,9 +49,9 @@
  * @return Positioned bounding boxes intersect
  */
 inline UBYTE bb_intersects(rect16_t *bb_a, upoint16_t *offset_a, rect16_t *bb_b, upoint16_t *offset_b) {
     if ((offset_b->x + bb_b->left   > offset_a->x + bb_a->right) ||
-        (offset_b->x + bb_b->right  < offset_a->x + bb_a->left)) return FALSE;    
+        (offset_b->x + bb_b->right  < offset_a->x + bb_a->left)) return FALSE;
     if ((offset_b->y + bb_b->top    > offset_a->y + bb_a->bottom) ||
         (offset_b->y + bb_b->bottom < offset_a->y + bb_a->top)) return FALSE;
     return TRUE;
 }
@@ -61,17 +63,17 @@
  * @param ty Top tile
  * @return Tile value, 0 if no collisions, COLLISION_ALL if out of bounds
  */
 inline UBYTE tile_at(UBYTE tx, UBYTE ty) {
-    if ((tx < image_tile_width) && (ty < image_tile_height)) 
+    if ((tx < image_tile_width) && (ty < image_tile_height))
         return ReadBankedUBYTE(collision_ptr + (ty * (UINT16)image_tile_width) + tx, collision_bank);
-    return COLLISION_ALL;
+    return (scene_transition_enabled)? 0: COLLISION_ALL;
 }
 
 /**
  * Test for a tile matching mask in a vertical range from ty_start to ty_end at column tx.
  * Updates globals tile_hit_x and tile_hit_y which can be read afterwards to determine which tile matched
- * 
+ *
  * @param tile_mask Tile bitmask to match
  * @param tx Tile x-coordinate
  * @param ty_start Starting tile y-coordinate
  * @param ty_end Ending tile y-coordinate
