Index: src\core\data_manager.c
===================================================================
--- src\core\data_manager.c
+++ src\core\data_manager.c
@@ -14,8 +14,10 @@
 #include "ui.h"
 #include "palette.h"
 #include "data/spritesheet_none.h"
 #include "data/data_bootstrap.h"
+#include "data/states_defines.h"
+#include "scene_transition.h"
 #include "macro.h"
 
 #define ALLOC_BKG_TILES_TOWARDS_SPR
 
@@ -58,9 +60,13 @@
     scene_stack_ptr = scene_stack;
 }
 
 void load_bkg_tileset(const tileset_t* tiles, UBYTE bank) BANKED {
-    if ((!bank) && (!tiles)) return;
+    #ifdef DISABLE_TILESET_LOAD_ON_TRANSITION
+        if (((!bank) && (!tiles)) || is_transitioning_scene) return;    
+    #else
+        if ((!bank) && (!tiles)) return;
+    #endif
 
     UWORD n_tiles = ReadBankedUWORD(&(tiles->n_tiles), bank);
 
     // load first background chunk, align to zero tile
@@ -159,14 +165,37 @@
 }
 
 UBYTE do_load_palette(palette_entry_t * dest, const palette_t * palette, UBYTE bank) BANKED {
     UBYTE mask = ReadBankedUBYTE(&palette->mask, bank);
-    dest;
-    palette;
-    bank;
+    palette_entry_t * sour = palette->cgb_palette;
+    for (UBYTE i = mask; (i); i >>= 1, dest++) {
+        if ((i & 1) == 0) continue;
+        MemcpyBanked(dest, sour, sizeof(palette_entry_t), bank);
+        sour++;
+    }
     return mask;
 }
 
+inline void load_bkg_palette(const palette_t * palette, UBYTE bank) {
+    UBYTE mask = do_load_palette(BkgPalette, palette, bank);
+    DMG_palette[0] = ReadBankedUBYTE(palette->palette, bank);
+#ifdef SGB
+    if (_is_SGB) {
+        UBYTE sgb_palettes = SGB_PALETTES_NONE;
+        if (mask & 0b00110000) sgb_palettes |= SGB_PALETTES_01;
+        if (mask & 0b11000000) sgb_palettes |= SGB_PALETTES_23;
+        SGBTransferPalettes(sgb_palettes);
+    }
+#endif
+}
+
+inline void load_sprite_palette(const palette_t * palette, UBYTE bank) {
+    do_load_palette(SprPalette, palette, bank);
+    UWORD data = ReadBankedUWORD(palette->palette, bank);
+    DMG_palette[1] = (UBYTE)data;
+    DMG_palette[2] = (UBYTE)(data >> 8);
+}
+
 UBYTE load_scene(const scene_t * scene, UBYTE bank, UBYTE init_data) BANKED {
     UBYTE i;
     scene_t scn;
 
@@ -184,14 +213,19 @@
 
     collision_bank  = scn.collisions.bank;
     collision_ptr   = scn.collisions.ptr;
 
+#ifndef DISABLE_UI_TILESET_LOAD
     // Load UI tiles, they may be overwritten by the following load_background()
     ui_load_tiles();
+#endif
 
     // Load background + tiles
     load_background(scn.background.ptr, scn.background.bank);
 
+    load_bkg_palette(scn.palette.ptr, scn.palette.bank);
+    load_sprite_palette(scn.sprite_palette.ptr, scn.sprite_palette.bank);
+
     // Copy parallax settings
     memcpy(&parallax_rows, &scn.parallax_rows, sizeof(parallax_rows));
     if (scn.parallax_rows[0].next_y == 0) {
         scene_LCD_type = (scene_type == SCENE_TYPE_LOGO) ? LCD_fullscreen : LCD_simple;
@@ -199,19 +233,27 @@
         scene_LCD_type = LCD_parallax;
     }
 
     scroll_x_min = scn.scroll_bounds.left;
-    scroll_x_max = scn.scroll_bounds.right;
+    scroll_x_max = ((image_width < SCREENWIDTH)? 0: scn.scroll_bounds.right + TILE_TO_PX(scroll_right_margin));
     scroll_y_min = scn.scroll_bounds.top;
-    scroll_y_max = scn.scroll_bounds.bottom;
+    scroll_y_max = ((image_height < SCREENHEIGHT)? 0: scn.scroll_bounds.bottom + TILE_TO_PX(scroll_bottom_margin));
 
     if (scene_type != SCENE_TYPE_LOGO) {
         // Load player
-        PLAYER.sprite = scn.player_sprite;
-        UBYTE n_loaded = load_sprite(PLAYER.base_tile = 0, scn.player_sprite.ptr, scn.player_sprite.bank);
-        allocated_sprite_tiles = (n_loaded > scn.reserve_tiles) ? n_loaded : scn.reserve_tiles;
-        load_animations(scn.player_sprite.ptr, scn.player_sprite.bank, ANIM_SET_DEFAULT, PLAYER.animations);
-        load_bounds(scn.player_sprite.ptr, scn.player_sprite.bank, &PLAYER.bounds);
+    #ifdef DISABLE_PLAYER_SPRITE_LOAD_ON_TRANSITION
+        if (is_transitioning_scene){
+            allocated_sprite_tiles = player_sprite_len;
+        } else {
+    #endif
+            PLAYER.sprite = scn.player_sprite;
+            UBYTE n_loaded = load_sprite(PLAYER.base_tile = 0, scn.player_sprite.ptr, scn.player_sprite.bank);
+            allocated_sprite_tiles = player_sprite_len = (n_loaded > scn.reserve_tiles) ? n_loaded : scn.reserve_tiles;
+            load_animations(scn.player_sprite.ptr, scn.player_sprite.bank, ANIM_SET_DEFAULT, PLAYER.animations);
+            load_bounds(scn.player_sprite.ptr, scn.player_sprite.bank, &PLAYER.bounds);
+    #ifdef DISABLE_PLAYER_SPRITE_LOAD_ON_TRANSITION
+        }
+    #endif
     } else {
         // no player on logo, but still some little amount of actors may be present
         PLAYER.base_tile = allocated_sprite_tiles = 0x68;
         PLAYER.sprite = spritesheet_none_far;
@@ -305,9 +347,12 @@
     if (triggers_len != 0) {
         MemcpyBanked(&triggers, scn.triggers.ptr, sizeof(trigger_t) * triggers_len, scn.triggers.bank);
     }
 
+    scene_transition_reset();
     scroll_reset();
+    scroll_offset_x = 0;
+    scroll_offset_y = 0;
     trigger_reset();
 
     emote_actor = NULL;
 
