Index: src\core\meta_tiles.c
===================================================================
--- src\core\meta_tiles.c
+++ src\core\meta_tiles.c
@@ -11,8 +11,9 @@
 #include "gbs_types.h"
 #include "actor.h"
 #include "data/game_globals.h"
 #include "data_manager.h"
+#include "scene_transition.h"
 #include "data/states_defines.h"
 
 #if METATILE_SIZE == METATILE_SIZE_16
 #define METATILE_TILE_WIDTH 2
@@ -184,14 +185,14 @@
 #else
         MemcpyBanked(sram_map_data + METATILE_MAP_OFFSET(dest_x, current_y), tilemap_ptr + (UWORD)(((source_y + i) * bkg.width) + source_x), width, bkg.tilemap.bank);
 #endif
         if (commit){
-            bkg_address_offset = ((UWORD)get_bkg_xy_addr(dest_x & 31, current_y & 31)) - 0x9800;
+            bkg_address_offset = ((UWORD)get_bkg_xy_addr((dest_x + bkg_offset_x) & 31, (dest_y + bkg_offset_y) & 31)) - 0x9800;
             load_metatile_row(metatile_ptr, dest_x, current_y, width, metatile_bank);
             #ifdef CGB
                 if (_is_CGB) {
                     VBK_REG = 1;
-                    bkg_address_offset = ((UWORD)get_bkg_xy_addr(dest_x & 31, current_y & 31)) - 0x9800;
+                    bkg_address_offset = ((UWORD)get_bkg_xy_addr((dest_x + bkg_offset_x) & 31, (dest_y + bkg_offset_y) & 31)) - 0x9800;
                     load_metatile_row(metatile_attr_ptr, dest_x, current_y, width, metatile_attr_bank);
                     VBK_REG = 0;
                 }
             #endif
@@ -247,9 +248,9 @@
         if (_is_CGB) {
             VBK_REG = 1;
         #if METATILE_SIZE == METATILE_SIZE_16
             tile_map_offset = TILE_MAP_OFFSET(tile_id, x, y);
-            bkg_address_offset = ((UWORD)get_bkg_xy_addr(x & 31, y & 31)) - 0x9800;				
+            bkg_address_offset = ((UWORD)get_bkg_xy_addr((x + bkg_offset_x) & 31, (y + bkg_offset_y) & 31)) - 0x9800;				
             set_vram_byte((UBYTE*)(0x9800 + bkg_address_offset), ReadBankedUBYTE(metatile_attr_ptr + tile_map_offset, metatile_attr_bank));
             tile_map_offset++;
             bkg_address_offset = (bkg_address_offset & 0xFFE0) + ((bkg_address_offset + 1) & 31);
             set_vram_byte((UBYTE*)(0x9800 + bkg_address_offset), ReadBankedUBYTE(metatile_attr_ptr + tile_map_offset, metatile_attr_bank));
@@ -266,9 +267,9 @@
         }
     #endif
     #if METATILE_SIZE == METATILE_SIZE_16
         tile_map_offset = TILE_MAP_OFFSET(tile_id, x, y);
-        bkg_address_offset = ((UWORD)get_bkg_xy_addr(x & 31, y & 31)) - 0x9800;				
+        bkg_address_offset = ((UWORD)get_bkg_xy_addr((x + bkg_offset_x) & 31, (y + bkg_offset_y) & 31)) - 0x9800;				
         set_vram_byte((UBYTE*)(0x9800 + bkg_address_offset), ReadBankedUBYTE(metatile_ptr + tile_map_offset, metatile_bank));
         tile_map_offset++;
         bkg_address_offset = (bkg_address_offset & 0xFFE0) + ((bkg_address_offset + 1) & 31);
         set_vram_byte((UBYTE*)(0x9800 + bkg_address_offset), ReadBankedUBYTE(metatile_ptr + tile_map_offset, metatile_bank));
