Index: src\core\collision.c
===================================================================
--- src\core\collision.c
+++ src\core\collision.c
@@ -3,8 +3,9 @@
 #include "collision.h"
 #include "meta_tiles.h"
 #include "math.h"
 #include "data/states_defines.h"
+#include "scene_transition.h"
 
 UBYTE tile_hit_x = 0;
 UBYTE tile_hit_y = 0;
 
@@ -30,9 +31,9 @@
             tile_ptr -= image_tile_width;
         }
         if (tile_hit_y >= image_tile_height) {
           SWITCH_ROM(_save);
-          return COLLISION_ALL & tile_mask ? COLLISION_ALL : 0;
+          return (!scene_transition_enabled && (COLLISION_ALL & tile_mask)) ? COLLISION_ALL : 0;
         }
     }
     SWITCH_ROM(_save);
     return 0;
@@ -42,9 +43,9 @@
 	tile_hit_x = tx;
     tile_hit_y = ty_start;
 	
     if (tile_hit_x >= image_tile_width || tile_hit_y >= image_tile_height) {
-      return COLLISION_ALL & tile_mask ? COLLISION_ALL : 0;
+      return (!scene_transition_enabled && (COLLISION_ALL & tile_mask)) ? COLLISION_ALL : 0;
     }
 	if (metatile_collision_bank) {
 #if METATILE_SIZE == METATILE_SIZE_16
         UBYTE metatile_x_offset = METATILE_X_OFFSET(tx);
@@ -69,9 +70,9 @@
 			} else {
 				tile_hit_y--;
 			}
 			if (tile_hit_y >= image_tile_height) {
-				return COLLISION_ALL & tile_mask ? COLLISION_ALL : 0;
+				return (!scene_transition_enabled && (COLLISION_ALL & tile_mask)) ? COLLISION_ALL : 0;
 			}
 		}
 		return 0;
 	}
@@ -100,9 +101,9 @@
             tile_ptr--;
         }
         if (tile_hit_x >= image_tile_width) {
           SWITCH_ROM(_save);
-          return COLLISION_ALL & tile_mask ? COLLISION_ALL : 0;
+          return (!scene_transition_enabled && (COLLISION_ALL & tile_mask)) ? COLLISION_ALL : 0;
         }                
     }
     SWITCH_ROM(_save);
     return 0;
@@ -111,9 +112,9 @@
 UBYTE tile_col_test_range_x(UBYTE tile_mask, UBYTE ty, UBYTE tx_start, UBYTE tx_end) BANKED {
 	tile_hit_x = tx_start;
     tile_hit_y = ty;
     if (tile_hit_x >= image_tile_width || tile_hit_y >= image_tile_height) {
-      return COLLISION_ALL & tile_mask ? COLLISION_ALL : 0;
+      return (!scene_transition_enabled && (COLLISION_ALL & tile_mask)) ? COLLISION_ALL : 0;
     }
 	if (metatile_collision_bank) {
         UWORD metatile_y_offset = METATILE_Y_OFFSET(ty);
 #if METATILE_SIZE == METATILE_SIZE_16
@@ -138,9 +139,9 @@
             } else {
                 tile_hit_x--;
             }
             if (tile_hit_x >= image_tile_width) {
-                return COLLISION_ALL & tile_mask ? COLLISION_ALL : 0;
+                return (!scene_transition_enabled && (COLLISION_ALL & tile_mask)) ? COLLISION_ALL : 0;
             }
         }
         return 0;
 	}
